docs/rules/object-curly-newline.md

```diff
@@ -19,6 +19,7 @@ Or an object option:
 
 * `"multiline": true` (default) requires line breaks if there are line breaks inside properties or between properties
 * `"minProperties"` requires line breaks if the number of properties is more than the given integer
+* `"consistent"` requires that either both curly braces, or neither, directly enclose newlines
```

Comment:
I read the issue, but I'm still not sure I understand the changes that it's proposing. Based on https://github.com/eslint/eslint/issues/6488#issuecomment-256178368, the following code should report an error:

```js
/* eslint object-curly-newline: [2, {"minProperties": 3, "multiline": true}] */

var oneM = {
  a: 1 // there is only one property and minProperties is 3, so an error is reported
};
```

But this should not report an error:

```js
/* eslint object-curly-newline: [2, {"minProperties": 3, "multiline": true, "consistent": true}] */

var oneM = {
  a: 1 // there is still only one property and minProperties is 3. Did something change?
};
```

Based on these examples, my understanding is that enabling the `consistent` option is implicitly changing the behavior of the `minProperties` option. With the `consistent` option disabled, the rule reports an object if it contains fewer than `minProperties` linebreaks, but with the `consistent` option is enabled, the rule does not report an object like that. Is this correct?

If so, it seems like a poor design choice in my opinion; I would not expect that enabling one option would implicitly change the behavior of another option.

cc @ljharb